### PR TITLE
Assign area labels to PRs based on changed paths

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR will enable a GitHub action to automatically assign area labels to PRs based on changed paths. This will help triage and changelog reporting.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# This file is a map of labels to repo glob paths. See https://github.com/marketplace/actions/labeler
+# for more details.
+
+"area: dds":
+  - any: ["/packages/dds/**", "!/packages/dds/sequence/**"]
```
Comment (anthony-murphy) [inline .github/labeler.yml:5]: i'd rather have both tags. does it support that?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# Any change to BREAKING.md or to any api-report path in the repo
+"breaking change":
+  - ./BREAKING.md
+  - api-report/*
+
+documentation:
+  - /docs/content/**
+
+testing:
```
Comment (anthony-murphy) [inline .github/labeler.yml:49]: can we add an "area: test" label?

**Review by anthony-murphy — APPROVED** [review]

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+"area: tools":
+  - /tools/**
+
+"area: website":
+  - any: ["/docs/**", "!/docs/content/**"]
+
+# Any change to BREAKING.md or to any api-report path in the repo
+"breaking change":
+  - ./BREAKING.md
+  - api-report/*
```
Comment (anthony-murphy) [inline .github/labeler.yml:44]: will this trigger for any api-report folder? even the nested ones?

Comment (tylerbutler) [reply .github/labeler.yml:44]: That's my understanding from the docs. I'll update if it doesn't work as expected.

File: .github/labeler.yml (hunk 1 above)
Comment (tylerbutler) [reply .github/labeler.yml]: Updated.

File: .github/labeler.yml (hunk 2 above)
Comment (tylerbutler) [reply .github/labeler.yml]: Done.

File: .github/labeler.yml (hunk 1 above)
Comment (anthony-murphy) [reply .github/labeler.yml]: if it doesn't i think i'd prefer the general dds label for now

**Pull request merged**